tools/ci-scripts/windows/Install-Boost.ps1

```diff
@@ -122,14 +122,18 @@ function Install-BoostComponent([string]$Component)
             -PackageId $packageId `
             -InstallDir $workDir `
             -PackageVersion $Version
-
+        if($Version < 1.66){
```

Comment:
Comparing versions as either numbers or strings will not work correctly. A numeric sort won't handle versions like 1.2.3, and string sorting fails on versions like 10.1 compared to 2.8.

Try casting the version to `System.Version` and comparing those:

```powershell
if ([System.Version]$Version -lt [System.Version]"1.66") {
```